app/controllers/bid_acceptances_controller.rb

```diff
@@ -1,9 +1,12 @@
 class BidAcceptancesController < ApplicationController
-	before_action :authenticate_broker_or_forward_freight!
+	before_action :authenticate_business_user!
 
   def create
     @bid = Bid.find(params[:bid_id])
     @bid.toggle_accept(@bid.accepted)
+    
+    @bid.accepted ? @bid.job.update(agent_id: @bid.bidder.id) : @bid.job.update(agent_id: nil)
```

Comment:
```self.agent_id = accepted ? bidder.id : nil ``` this would try to update the agent_id of Bid, but we should update the agent_id of Job, right? Would doing something like ```self.job.agent_id = accepted ? bidder.id : nil``` work?